### ISSUE TITLE
PropertyReplacer to be changed to fail hard if is not all property tokens are substituted 

### ISSUE THREAD

**Issue by RobAustin** [OpenHFT/Chronicle-Core#139]
The net.openhft.chronicle.bytes.util.PropertyReplacer is used to replace tokens like ${<system.property>} in strings, these strings are typically used as YAML configuration in many of our applications, we should system.exit() [ and produce an error message ]  if all the tokens are not substituted,  as users can easily miss configure the system properties leaving the application to continue on blindly ( with the wrong configuration ). Currently the net.openhft.chronicle.bytes.util.PropertyReplacer does not system exit and it is currently too easy to start the application with properties miss configured.

**Comment by minborg** [comment]
or throw an Exception?

**Comment by JerryShea** [comment]
Yes throwing an exception please. System.exit is much too intrusive

**Comment by RobAustin** [comment]
I agree.
